src/paperless/views.py

```diff
@@ -171,3 +174,70 @@ class ApplicationConfigurationViewSet(ModelViewSet):
 
     serializer_class = ApplicationConfigurationSerializer
     permission_classes = (IsAuthenticated,)
+
+
+class DisconnectSocialAccountView(GenericAPIView):
+    """
+    Disconnects a social account provider from the user account
+    """
+
+    permission_classes = [IsAuthenticated]
+
+    def post(self, request, *args, **kwargs):
+        user = self.request.user
+
+        user.socialaccount_set.get(pk=request.data["id"]).delete()
+
+        return Response(
+            request.data["id"],
```

Comment:
The security warning seems valid, also will this validate the account? Perhaps something like:

```suggestion
        account = user.socialaccount_set.get(pk=request.data["id"])
        if account is not None:
            account.delete()
            return Response(
                account.id,
            )
        else:
            return HttpResponseBadRequest(
                "Social account not found",
```

Reply:
Isn't get throwing an exception if it cannot find exactly one object? Than my version be save but indeed doesn't return the best error response. How about:

```
try:
   account = user.socialaccount_set.get(pk=request.data["id"])
   account.delete()
   return Response(account.id)
except SocialAccount.DoesNotExist:
   return HttpResponseBadRequest("Social account not found")
```

Reply:
Sure, same idea I think